internal/provider/provider.go

```diff
@@ -445,6 +445,14 @@ func configure(ctx context.Context, provider *schema.Provider, d *schema.Resourc
 		terraformVersion = "0.11+compatible"
 	}
 
+	// Terraform 0.12.31 will download beta version of the provider as the latest available.
+	// See https://github.com/hashicorp/terraform/issues/36586.
+	// This check must be removed before the GA release.
+	switch {
+	case terraformVersion == "0.11+compatible", strings.HasPrefix(terraformVersion, "0.12."):
+		return nil, sdkdiag.AppendErrorf(diags, "unsupported Terraform version: %s", terraformVersion)
```

Comment:
Overlooked this on the first pass, but since this is displayed in a diagnostic we could potentially get away with a longer message than what is typical for go errors. 

Perhaps we can include some additional context indicating how to pin versions and/or that `v0.12` will still be supported, just not until after GA?